// Compile via python -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. Client.proto 

syntax = "proto3";

package Client;

// Status values
enum Status {
  SUCCESS = 1;
  ERROR = 2;
}

// Register Room
message RegisterRoomRequest{
    string address = 1;
    string username = 2;
}
message RegisterRoomResponse{
    Status status = 1;
}

// Load a song
message LoadSongRequest{
    string name = 1;
    bytes AudioData = 2;
}
message LoadSongResponse{
}

// Start Song
message StartSongRequest {
    double time = 1;
    double offset = 2;
}
message StartSongResponse {
}

// Pause Song
message PauseSongRequest {
    double time = 1;
}
message PauseSongResponse {
}

service Client {
    rpc RegisterRoom(RegisterRoomRequest) returns (RegisterRoomResponse);
    rpc LoadSong(LoadSongRequest) returns (LoadSongResponse);
    rpc StartSong(StartSongRequest) returns (StartSongResponse);
    rpc PauseSong(PauseSongRequest) returns (PauseSongResponse);
}